ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/validate/ConflictCheckWithCommittedImpl.scala

```diff
@@ -149,4 +158,96 @@ private[validate] class ConflictCheckWithCommittedImpl(
     }
   }
 
+  private def sameContractData(
+      actual: (Value.VersionedContractInstance, Timestamp),
+      provided: DisclosedContract,
+  )(implicit loggingContext: LoggingContext): Boolean = {
+    // TODO DPP-1026: Note: Not validating transaction version or agreement text here, because we're making those up
```

Comment:
I'm not quite sure the normalization is in the right place. Let’s compare to how things work for Daml transactions:
1. Users pass in unnormalized daml-lf values via the ledger API as part of the commands.
2. Daml Engine takes those values and produces a transaction in those.
3. The resulting transaction is then sent fully normalized over the write service.

Here you are sending unnormalized data over the write service and then normalizing here. My suggestion would be that we do the normalization (using `normalizeValue` as Rémy suggested) before calling the write service and always send normalized values here.

There is also a question of validation of those disclosed contracts: It looks like currently you pass around all disclosed contracts regardless of whether they are used or not. This means that those contracts may not even have type-correct payloads (or maybe even potentially invalid template ids, not quite sure if you check for that).

My expectation was that we would only send used disclosed contracts and normalize & typecheck them.

Reply:
resolving in favor of https://github.com/digital-asset/daml/issues/13864